### PR TITLE
Feat/onboarding username

### PR DESCRIPTION
**Pull request by agustif**
## What does this PR do?
Adds - username text field (from profile settings) made by @alannnc in getting started onboarding, if username is not already set.

Necessary for the new magic signup link, in case you try to purchase a premium username and dont finish checkout, you end up without a username but logged In and in onboarding.

Goes along this other PR in website private repo for magic signup login link https://github.com/calcom/website/pull/172

**Environment**: main

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?
- [ ] run maildev -s 587 , maybe need to change .env SMTP_HOST to localhost or `0.0.0.0`
- [ ] checkout This PR #3580 on monorepo, and  PR https://github.com/calcom/website/pull/172 on website, test the new signup with a premium username, but cancel, you'll see the email in localhost:1080 to login.
- [ ] Login, and the username text field should appear as first step of getting started

If no username, onboarding should look like this
image

### CODE REVIEW

File: apps/web/components/ui/UsernameAvailability/PremiumTextfield.tsx [hunk 1]
```diff
@@ -246,7 +246,7 @@ const PremiumTextfield = (props: ICustomUsernameProps) => {
       )}
 
       {(usernameIsAvailable || premiumUsername) && currentUsername !== inputUsernameValue && (
-        <div className="mt-2 flex justify-end md:hidden">
+        <div className="mt-2 flex justify-end sm:hidden">
```
Comment (agustif) [inline apps/web/components/ui/UsernameAvailability/PremiumTextfield.tsx:249]: @alannnc made this changes so I wouldn't get double buttons in some resolutions

File: apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx [hunk 2]
```diff
@@ -151,7 +151,7 @@ const UsernameTextfield = (props: ICustomUsernameProps) => {
       {markAsError && <p className="mt-1 text-xs text-red-500">Username is already taken</p>}
 
       {usernameIsAvailable && currentUsername !== inputUsernameValue && (
-        <div className="mt-2 flex justify-end md:hidden">
+        <div className="mt-2 flex justify-end sm:hidden">
```
Comment (agustif) [inline apps/web/components/ui/UsernameAvailability/UsernameTextfield.tsx:154]: same here double buttons with md

**Comment by PeerRich** [comment]
CleanShot 2022-07-30 at 10 19 27@2x
is the massive space between label and input intended?

File: apps/web/pages/getting-started.tsx [hunk 3]
```diff
@@ -205,22 +210,15 @@ export default function Onboarding(props: inferSSRProps<typeof getServerSideProp
...
+          return createEventType.mutate(event);
+        })
+      );
     }
-    // Write default timeformat to localStorage
-    const browserTimeFormat = isBrowserLocale24h() ? 24 : 12;
-
     await updateUser({
       completedOnboarding: true,
-      timeFormat: browserTimeFormat,
```
Comment (zomars) [inline apps/web/pages/getting-started.tsx:223]: Is this intended @agustif ?

File: apps/web/pages/getting-started.tsx [hunk 4]
```diff
@@ -301,7 +299,7 @@ export default function Onboarding(props: inferSSRProps<typeof getServerSideProp
                       token: values.token,
                     }),
                     headers: {
-                      "Content-Type": "application/json",
+                      "Content-Type": "applishcation/json",
```
Comment (zomars) [inline apps/web/pages/getting-started.tsx:302]:
```suggestion
                      "Content-Type": "application/json",
```

File: apps/web/pages/getting-started.tsx [hunk 5]
```diff
@@ -371,8 +383,8 @@ export default function Onboarding(props: inferSSRProps<typeof getServerSideProp
                 <TimezoneSelect
                   id="timeZone"
                   value={selectedTimeZone}
-                  onChange={({ value }) => setSelectedTimeZone(value)}
-                  className="mt-1 block w-full rounded-md border-gray-300 text-sm focus:border-blue-500 focus:ring-blue-500"
+                  onChange={({ value }: { value: string }) => setSelectedTimeZone(value)}
```
Comment (zomars) [inline apps/web/pages/getting-started.tsx:386]:
```suggestion
                  onChange={({ value }) => setSelectedTimeZone(value)}
```

This should be inferred AFAIK

File: apps/web/pages/getting-started.tsx [hunk 6]
```diff
@@ -522,7 +535,6 @@ export default function Onboarding(props: inferSSRProps<typeof getServerSideProp
   /** End Onboarding Steps */
 
   useEffect(() => {
-    detectStep();
```
Comment (zomars) [inline apps/web/pages/getting-started.tsx:525]: This comes from the server now

File: apps/web/pages/getting-started.tsx [hunk 7]
```diff
@@ -641,12 +654,41 @@ export async function getServerSideProps(context: NextPageContext) {
...
+        },
+      },
+      schedules: {
+        where: {
+          userId: session.user.id,
+        },
+        select: {
+          id: true,
+        },
+      },
```
Comment (zomars) [inline apps/web/pages/getting-started.tsx:691]: Now getting this data in one query instead of 3

File: apps/web/pages/getting-started.tsx [hunk 8]
```diff
@@ -662,59 +704,18 @@ export async function getServerSideProps(context: NextPageContext) {
...
 
   return {
     props: {
       session,
       user,
-      integrations,
-      connectedCalendars,
-      eventTypes,
-      schedules,
+      initialStep: hasSchedules ? (hasConfigureCalendar ? 2 : 3) : 0,
```
Comment (zomars) [inline apps/web/pages/getting-started.tsx:718]: Only passing necessary data to the frontend. Reducing HTML size and possible data leakage

File: packages/prisma/schema.prisma [hunk 9]
```diff
@@ -75,6 +75,7 @@ model EventType {
   currency                String                  @default("usd")
   slotInterval            Int?
   metadata                Json?
+  /// @zod.custom(imports.successRedirectUrl)
```
Comment (zomars) [inline packages/prisma/schema.prisma:78]: Validate `successRedirectUrl` as early as possible

File: packages/trpc/server/routers/viewer/eventTypes.tsx [hunk 10]
```diff
@@ -102,9 +86,20 @@ const EventTypeUpdateInput = _EventTypeModel
...
+          id: true,
+          title: true,
+          description: true,
+          length: true,
+          schedulingType: true,
+          slug: true,
+          hidden: true,
+          metadata: true,
         },
       });
```
Comment (zomars) [inline packages/trpc/server/routers/viewer/eventTypes.tsx:104]: This endpoint was wrong and unused. Fixed to deprecate the last non-trpc query in onboarding.

File: packages/trpc/server/routers/viewer/eventTypes.tsx [hunk 11]
```diff
@@ -269,7 +264,6 @@ export const eventTypesRouter = createProtectedRouter()
         hashedLink,
         ...rest
       } = input;
-      assertValidUrl(input.successRedirectUrl);
```
Comment (zomars) [inline packages/trpc/server/routers/viewer/eventTypes.tsx:272]: This now happens at zod

**Review by zomars — APPROVED** [review]
Merging to test on main. Will revert if necessary

**Pull request merged**